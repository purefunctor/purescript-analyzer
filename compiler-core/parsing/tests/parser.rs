use test_each_file::test_each_file;

test_each_file! { in "./compiler-core/parsing/tests/parser" => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, errors) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    insta::assert_debug_snapshot!((node, errors));
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as lossless => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, _) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    assert_eq!(node.to_string(), content);
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as stability => |content: &str| {
    let lexed = lexing::lex(content);
    for index in 0..lexed.len() - 1 {
        let partial = lexed.text_in_range(0..index + 1);
        let lexed = lexing::lex(partial);
        let tokens = lexing::layout(&lexed);
        let (parsed, _) = parsing::parse(&lexed, &tokens);
        let node = parsed.syntax_node();
        assert_eq!(node.to_string(), partial);
    }
}}
