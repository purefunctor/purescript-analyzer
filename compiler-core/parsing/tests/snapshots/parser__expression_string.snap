---
source: compiler-core/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..69
      ModuleHeader@0..29
        MODULE@0..6 "module"
        ModuleName@6..23
          Annotation@6..7
            TEXT@6..7 " "
          UPPER@7..23 "ExpressionString"
        Annotation@23..24
          TEXT@23..24 " "
        WHERE@24..29 "where"
      LAYOUT_START@29..29 ""
      ModuleImports@29..29
      ModuleStatements@29..68
        ValueEquation@29..47
          Annotation@29..31
            TEXT@29..31 "\n\n"
          LOWER@31..37 "string"
          FunctionBinders@37..37
          Unconditional@37..47
            Annotation@37..38
              TEXT@37..38 " "
            EQUAL@38..39 "="
            WhereExpression@39..47
              ExpressionString@39..47
                Annotation@39..40
                  TEXT@39..40 " "
                STRING@40..47 "\"hello\""
        LAYOUT_SEPARATOR@47..47 ""
        ValueEquation@47..68
          Annotation@47..48
            TEXT@47..48 "\n"
          LOWER@48..54 "string"
          FunctionBinders@54..54
          Unconditional@54..68
            Annotation@54..55
              TEXT@54..55 " "
            EQUAL@55..56 "="
            WhereExpression@56..68
              ExpressionString@56..68
                Annotation@56..57
                  TEXT@56..57 " "
                RAW_STRING@57..68 "\"\"\"hello\"\"\""
      LAYOUT_END@68..68 ""
      Annotation@68..69
        TEXT@68..69 "\n"
      END_OF_FILE@69..69 ""
    ,
    [],
)
